wire-runtime/src/main/java/com/squareup/wire/ReflectiveMessageAdapter.java

```diff
@@ -280,44 +280,25 @@ private int getSerializedSize(int tag, Object value, Datatype datatype) {
    */
   private int getSerializedSizeNoTag(Object value, Datatype datatype) {
     switch (datatype) {
-      case INT32: return ProtoWriter.int32Size((Integer) value);
-      case INT64: case UINT64: return ProtoWriter.varint64Size((Long) value);
-      case UINT32: return ProtoWriter.varint32Size((Integer) value);
-      case SINT32: return ProtoWriter.varint32Size(ProtoWriter.zigZag32((Integer) value));
-      case SINT64: return ProtoWriter.varint64Size(ProtoWriter.zigZag64((Long) value));
-      case BOOL: return 1;
-      case ENUM: return getEnumSize((ProtoEnum) value);
-      case STRING:
-        int utf8Length = utf8Length((String) value);
-        return ProtoWriter.varint32Size(utf8Length) + utf8Length;
-      case BYTES:
-        int length = ((ByteString) value).size();
-        return ProtoWriter.varint32Size(length) + length;
+      case INT32: return TypeAdapter.INT32.serializedSize((Integer) value);
```

Comment:
All this stuff is just temporary until it's wired through more and these switches disappear completely!